### DISCUSSION TITLE
Can you add api for https://dev.freebox.fr/sdk/os/download/

### DISCUSSION THREAD

**Discussion by PierreCsn** [pymedusa/Medusa#8064] [Ideas & Feature requests]
Hello one major ISP in france has a downloader with API on the router. Any chance it can be added to medusa we have millions of user in France and usually, they are more likely to use services like medusa than Users from other ISPs

Here is a link to every API they provide : https://dev.freebox.fr/sdk/os/download/

The downloader is fairly efficient and can handle proper VPN connections so it would be pretty useful for people to be able to add this one as it s already useable instead of requiring an additional computer / server to handle downloads.

I did some research and I didn't see if any service that would be compatible already with those API .
I m available if any dev want to do some testing with me.
best regards

**Comment by froggy974** [comment]
It would be great ! 
because the transmission remote configuration of medusa does not work.
the correct adress to use is for transmission login is `http://mafreebox.freebox.fr:9091/transmission/rpc` but medusa probably add a "/" at the end, and transform this adress to `http://mafreebox.freebox.fr:9091/transmission/rpc/`   so it fail

**Comment by medariox** [comment]
Medusa doesn't add any trailing slash, you just need to configure it properly. In your case you just need to use `transmission` as RPC URL.

**Comment by froggy974** [comment]
it's possible that i made a mistake on medusa transmission settings, but i use the same settings in a windows transmision remoteclient and it's working
see screenshot :
![2020-05-10_16-30-33](https://user-images.githubusercontent.com/17068466/81502436-45394e80-92de-11ea-9911-d090f3becd26.png)
![2020-05-10_16-49-07](https://user-images.githubusercontent.com/17068466/81502438-47031200-92de-11ea-9a2f-abbb4722410d.png)

so it's probably on medusa side

**Comment by medariox** [comment]
Different software, different settings! Maybe try to do as I suggested and how it is written in the example?

**Comment by froggy974** [comment]
yes,  i tried every possible combination in RPC URL 
`transmission` 
`/transmission` 
`transmission/rpc` 
`/transmission/rpc` 

if a "freenaute" already managed to make it work with transmission, please let me know

**Comment by froggy974** [comment]
maybe it's also specific to the custom transmission settings on the ISP side.
the same problem occur with `$ transmission-remote` linux command

https://dev.freebox.fr/bugs/task/19407
